src/js/player.js

```diff
@@ -4290,6 +4300,69 @@ class Player extends Component {
     return !!this.isAudio_;
   }
 
+  /**
+   * Get the current audioOnlyMode state or set audioOnlyMode to true or false.
+   *
+   * Setting this to `true` will hide all player components except the control bar,
+   * as well as control bar components needed only for video.
+   *
+   * @param {boolean} [value]
+   *         The value to set audioOnlyMode to.
+   *
+   * @return {boolean}
+   *         True if audioOnlyMode is on, false otherwise.
+   */
+  audioOnlyMode(value) {
+    if (typeof value !== 'boolean' || value === this.audioOnlyMode_) {
+      return this.audioOnlyMode_;
+    }
+
+    if (value === true) {
+      // Update styling immediately so we can get the control bar's height
+      this.addClass('vjs-audio-only-mode');
+
+      const playerChildren = this.children();
+      const controlBar = this.getChild('ControlBar');
+      const controlBarHeight = controlBar && controlBar.height();
```

Comment:
non-blocking: We potentially want `currentHeight()` here which should give us the computed height and not the height that it was set to, which may not be the true height.
```suggestion
      const controlBarHeight = controlBar && controlBar.currentHeight();
```

In practice and based on how things are set up, this may not really matter here.